docs/source/concept_models/vivarium_census_synthdata/concept_model.rst

```diff
@@ -905,14 +909,29 @@ sources described above to create conditional samplers for first name
 and last name based on soundex.  Perhaps measure of success is to look
 at entropy of character n-gram distribution.
 
-To simulate naming after a parent or family member, have 5% of children 
-born in the simulation be named after a relative that matches their 
-sex. To do this: 
+To simulate naming after a parent or family member, we would like 5% of 
+children to have the same name as a parent. This can be separated into two 
+groups: 
+
+**People born in the simulation**
+
+For 5% of simulants, they will be assigned a name based on these steps: 
 
-#. If there is a tracked parent of the same sex, assign the same first name 
-#. If there is a relative of the same sex, assign the same first name. Note that this might be a relative of the reference person (i.e., a male child has a tracked mother, but that tracked mother has a tracked father) 
+#. Female simulants will have the same last name as their mother (who is known) 
+#. Male simulants, if the mother is the reference person and has an opposite-sex spouse, they will receive that name 
+#. Otherwise, if the mother has a male relative known in the simulation (either the mother is the reference person or is related to the reference person), they will be assigned a males relative's name. The relative can be selected at random. 
 #. If none of these are available, assign a random name 
 
+**People initialized in the simulation**
+
+For 5% of simulants, they will be assigned a name based on these steps: 
+
+#. For anyone who has a parent of the same sex known in the simulation (this means: ["reference person", "biological child"], ["reference person", "adopted child"], ["reference person", "parent"]) they have a 5% chance of having the same name as that parent 
+#. Note that for same sex couples, whoever is the reference person will pass their name instead of their spouse 
+#. For anyone with a parent of the opposite sex known in the simulation (same pairings), if that parent has a spouse in the simulation of the child's sex (["reference person", "opposite sex spouse"]), they will receive the same name of the spouse 
+#. For anyone else, if their parent has a relative of the correct sex in the simulation (either the parent is the reference person or is related to the reference person), they will receive a relative's name. If there multiple relative's at random 
```

Comment:
```suggestion
#. For anyone else, if they have a relative relationship attribute (any except "roommate/housemate" and "other nonrelative"), they are assigned the same first name as another randomly-selected person in the household who also has a relative relationship attribute and the same sex. If there is no such person, skip to the next step.
```

**I think this one could give us problems.** It does not define which relative is named first. Would be curious to hear engineering's take on it. Potentially there is some way to do all of this by manipulation of seeds, which wouldn't require there to be a specific "order."

Reply:
Hmm, I see the issue, we are hypothetically naming sims based on sims that might not have names yet. A couple of solutions: 
1. limit the matching names to one person per household which would fix this but is likely unrealistic as I would expect families likely to name kids after a parent or relative are also likely to do this for more children 
2. Define an order - oldest to youngest might make sense? So assign 95% randomly. If a household has multiple names that are meant to be matched, select the oldest to name first, then move down through the ages. If the oldest person is meant to be named the same as a yet unnamed younger person, they just get a random name 

I am going to implement 2. Let me know if you see an issue with that.